shared/eip712/eip712.go

```diff
@@ -0,0 +1,69 @@
+package eip712
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strings"
+)
+
+type EIP712Components struct {
+	V uint8    `json:"v"`
+	R [32]byte `json:"r"`
+	S [32]byte `json:"s"`
+}
+
+const EIP712Length = 65
+
+// Pretty print for EIP712Components
+func (e EIP712Components) String() {
+	fmt.Printf("EIP712 Components:\n")
+	fmt.Printf("V: %d\n", e.V)
+	fmt.Printf("R: %x\n", e.R)
+	fmt.Printf("S: %x\n", e.S)
+}
+
+// SanitizeHexInput checks for and strips a 0x prefix, returning an error if one is not found.
+// It returns the byte slice representing the decoded hex, or a decoding error
+func SanitizeHex(inp string) ([]byte, error) {
```

Comment:
I think we should just use <https://pkg.go.dev/github.com/ethereum/go-ethereum/common/hexutil>

Instead of SanitizeHex we can just call hexutil.Decode and bubble up any errors that occur.

The 0x prefix is mandatory in go-ethereum and for consistency we may as well require it ourselves.

So basically, delete this function